File: SWEA/D2/1961. 숫자 배열 회전/숫자 배열 회전.java
import java.util.Arrays;
import java.util.Scanner;
import java.util.stream.Collectors;
class Solution
{
    static int[][] rotate90(int[][] map, int n) {

        int[][] newMap = new int[n][n];
        for (int i = 0; i < n; i++) {
            Arrays.fill(newMap[i], 0);
        }

        // 90도 회전
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                newMap[i][j] = map[n - j - 1][i];
            }
        }

        return newMap;
    }
    
	public static void main(String args[]) throws Exception
	{
		Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();

        for (int t = 1; t <= T; t++) {
            int n = sc.nextInt();
            int[][] map = new int[n][n];

            // map 입력
            for (int i = 0; i < n; i++) {
                for (int j = 0; j < n; j++) {
                    map[i][j] = sc.nextInt();
                }
            }

            int[][] map90 = rotate90(map, n);
            int[][] map180 = rotate90(map90, n);
            int[][] map270 = rotate90(map180, n);

            System.out.printf("#%d\n", t);
            String re = "";
            for (int i = 0; i < n; i++) {
                re = Arrays.stream(map90[i])
                        .mapToObj(String::valueOf)
                        .collect(Collectors.joining());

                System.out.print(re + " ");

                re = Arrays.stream(map180[i])
                        .mapToObj(String::valueOf)
                        .collect(Collectors.joining());

                System.out.print(re + " ");

                re = Arrays.stream(map270[i])
                        .mapToObj(String::valueOf)
                        .collect(Collectors.joining());

                System.out.println(re);
            }
        }
	}
}

File: SWEA/D2/1961. 숫자 배열 회전/숫자 배열 회전.py

def rotate90(matrix, n):
    newMap = [[0] * n for _ in range(n)]

    for i in range(n):
        for j in range(n):
            newMap[i][j] = matrix[n - j - 1][i]

    return newMap


def rotate180(matrix, n):
    newMap = [[0] * n for _ in range(n)]
    for i in range(n):
        for j in range(n):
            newMap[i][j] = matrix[n - 1 - i][n - 1 - j]

    return newMap


def rotate270(matrix, n):
    newMap = [[0] * n for _ in range(n)]
    for i in range(n):
        for j in range(n):
            newMap[i][j] = matrix[j][n - 1 - i]

    return newMap


T = int(input())
for t in range(T):

    n = int(input())
    matrix = [list(map(int, input().split())) for _ in range(n)]
    map90 = rotate90(matrix, n)
    map180 = rotate180(matrix, n)
    map270 = rotate270(matrix, n)

    print(f"#{t+1}")
    for i in range(n):
        print("".join(map(str, map90[i])), end=' ')
        print("".join(map(str, map180[i])), end=' ')
        print("".join(map(str, map270[i])))